### PR TITLE
refactor-warnIfUnsavedChanges #1458

### LINKED ISSUE #1458: Refactor IDEView
#### Nature of issue?

- Existing feature enhancement

#### Feature enhancement details:

In order to work on making the web editor mobile friendly, @ghalestrilo suggested refactoring and breaking apart `IDEView.jsx`. I, personally, have wanted to refactor this component for a while, so I think now is a great time to work on this.

### PR DESCRIPTION
**Pull request by saijatin28**
RE: #1458 
This pull request aims at refactoring the **'warnIfUnsavedChanges'** function. I would request @catarak and @ghalestrilo to please take a look.
I would like to get some tips on how we could **split the function** further.

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

File: client/modules/IDE/pages/IDEView.jsx [hunk 1]
```diff
@@ -212,7 +212,7 @@ class IDEView extends React.Component {
         </Helmet>
         {this.props.toast.isVisible && <Toast />}
         <Nav
-          warnIfUnsavedChanges={this.warnIfUnsavedChanges}
+          warnIfUnsavedChanges={() => warnIfUnsavedChanges(this.props)}
```
Comment (catarak) [inline client/modules/IDE/pages/IDEView.jsx:215]: According to the React docs, you want to creating anonymous functions within render (I know there are many instances of this within the app, and I think they should all be fixed). Instead, I recommend adding a new named function to the IDEView class.

Comment (saijatin28) [reply client/modules/IDE/pages/IDEView.jsx]: Hey Cassie! I've replaced the anonymous function with a named function. Thoughts?

Comment (saijatin28) [reply client/modules/IDE/pages/IDEView.jsx]: Also Cassie, would you advice against using the .bind as well ?

**Comment by catarak** [comment]
Thanks for working on this! I added a couple more commits to get this 100% ready to be merged.

**Comment by saijatin28** [comment]
> Thanks for working on this! I added a couple more commits to get this 100% ready to be merged.

It's always my pleasure Cassie!
Alright! Thanks Cassie!

**Pull request merged**